src/event_emitter/base.cr

```diff
@@ -1,54 +1,67 @@
 module EventEmitter
-  class Base 
-    @all_handlers = [] of Proc(String, Any, Nil)
-    @channels = Hash(String, Array(Channel::Unbuffered(Any))).new
-
-    def all(&block : String, Any ->)
-      @all_handlers << ->(event : String, response : Any) {
-        block.call(event, response)
-      }
+  class Base
+    alias Event = NamedTuple(event: String | Regex, listener: Any ->, once: Bool, id: Int32)
```

Comment:
This seems a nice place to use the `record` macro! Or is there any advantage of using a `NamedTuple` instead of a `struct` that I'm not aware of?